### ISSUE TITLE
Campaign Reports | Inconsistent Data

### ISSUE THREAD

**Issue by BrandKits** [putyourlightson/craft-campaign#232]
### Describe the bug
Reporting data for Opened, Clicked and Unsubscribed is showing as different values in the Chart and Campaign Details summary. Plus the totals for opened is different in Locations and Devices to the Chart and Campaign Details summary. 

Clicking the value links shows inconsistent data also, and is truncated to 100 as per #231  preventing further investigation.

### To reproduce
Steps to reproduce the behaviour:
1. Send Campaign
2. Check Reports
3. Set Charts to Days
4. Compare

### Expected behaviour
Expect data to be consistent throughout, enabling confidence in the data and better understanding.

### Screenshots
<img width="1214" alt="campaign-reporting-inconsistencies" src="https://user-images.githubusercontent.com/38166209/121705905-08392000-cacd-11eb-8738-0f69b876f49e.png">

![image](https://user-images.githubusercontent.com/38166209/121706153-42a2bd00-cacd-11eb-8342-e375410ead87.png)

### Versions
- Plugin version: 1.19.2
- Craft version: 3.6.16

Thanks,
Paul

**Comment by bencroker** [comment]
Thanks Paul, we'll review the report stats and let you know what we find.

**Comment by BrandKits** [comment]
Thanks Ben, happy to provide CP access if you would like to see the reports from a number of campaigns.

**Comment by bencroker** [comment]
We've made some tweaks to the reporting, which was showing inconsistent stats for campaigns that were sent via multiple sendouts. It would be very helpful if you could test locally on your end and confirm whether this indeed fixes the inconsistencies you are seeing by requiring the `develop` branch:

```
composer require putyourlightson/craft-campaign:dev-develop
```

**Comment by bencroker** [comment]
Released in 1.20.0.

**Comment by bencroker** [comment]
@BrandKits had a chance to update yet?

**Comment by BrandKits** [comment]
Hi Ben, thanks for the update, keen to test but had to carve out a couple of weeks holiday, including second jab that wiped me out. Back on deck Monday and will test and report back.

**Comment by bencroker** [comment]
Closing due to inactivity.

**Comment by bencroker** [comment]
This has been finally fixed in https://github.com/putyourlightson/craft-campaign/commit/8cd94c0a204a53f7dc231a84e932c9bd5040bc23 and released in version 2.1.11, which includes a migration that will sync all campaign report data.
